front/src/modules/pathfinding/hook/usePathfinding.ts

```diff
@@ -0,0 +1,333 @@
+  const { rollingStock } = useStoreDataForRollingStockSelector();
+  // TODO TS2 : update this parts in margins and powerrestriction issues
+  // const powerRestrictions = useSelector(getPowerRestrictionRanges, isEqual);
+  // const allowances = useSelector(getAllowances, isEqual);
+  const initializerArgs = {
+    origin,
+    destination,
+    rollingStock,
+    pathSteps,
+    pathProperties,
+  };
+  const [pathfindingState, pathfindingDispatch] = useReducer(reducer, initializerArgs, init);
+
+  const [isPathfindingInitialized, setIsPathfindingInitialized] = useState(false);
+
+  const [postPathfindingBlocks] =
+    osrdEditoastApi.endpoints.postV2InfraByInfraIdPathfindingBlocks.useMutation();
+  const [postPathProperties] =
+    osrdEditoastApi.endpoints.postV2InfraByInfraIdPathProperties.useMutation();
+
+  const {
+    updatePathSteps,
+    // updatePowerRestrictionRanges,
+    // updateAllowances,
+  } = useOsrdConfActions();
+
+  const generatePathfindingParams = (): PostV2InfraByInfraIdPathfindingBlocksApiArg | null => {
+    setPathProperties(undefined);
+    return getPathfindingQuery({ infraId, rollingStock, origin, destination, pathSteps });
+  };
+
+  useEffect(() => {
+    if (isPathfindingInitialized) {
+      pathfindingDispatch({
+        type: 'VIAS_CHANGED',
+        params: {
+          origin,
+          destination,
+          rollingStock,
+        },
+      });
+    }
+  }, [vias]);
+
+  useEffect(() => {
+    if (isPathfindingInitialized) {
+      pathfindingDispatch({
+        type: 'PATHFINDING_PARAM_CHANGED',
+        params: {
+          origin,
+          destination,
+          rollingStock,
+        },
+      });
+    }
+  }, [origin, destination, rollingStock]);
+
+  useEffect(() => {
+    const startPathFinding = async () => {
+      if (!pathfindingState.running) {
+        pathfindingDispatch({ type: 'PATHFINDING_STARTED' });
+        const pathfindingInputV2 = generatePathfindingParams();
+        if (!pathfindingInputV2 || !infraId) {
+          dispatch(
+            setFailure({
+              name: t('pathfinding'),
+              message: t('pathfindingMissingParamsSimple'),
+            })
+          );
+          return;
+        }
+
+        try {
+          const pathfindingResult = await postPathfindingBlocks(pathfindingInputV2).unwrap();
+
+          if (pathfindingResult.status === 'success') {
+            const pathPropertiesParams: PostV2InfraByInfraIdPathPropertiesApiArg = {
+              infraId,
+              props: ['electrifications', 'geometry', 'operational_points'],
+              pathPropertiesInput: {
+                track_section_ranges: pathfindingResult.track_section_ranges,
+              },
+            };
+            const { electrifications, geometry, operational_points } =
+              await postPathProperties(pathPropertiesParams).unwrap();
+
+            if (electrifications && geometry && operational_points) {
```

Comment:
In operational studies, after a success pathfinding (TypeAndPathV2, ItineraryV2 when launched by map click), we have a `zoomToFeature` to fit both origin and destination in the map viewport. As we don't use either of these components for the stdcmv2 pathfinding know, we lost this feature and I think even though we don't want the path on the map, we should be able to see all path steps in the viewport when the pathfinding is successfull.
You can either add the function here (but might be redundant in operational studies or will need some refacto), or add it somewhere else in the stdcm v2 code.

Reply:
Will be done in another PR